clap_mangen/src/lib.rs

```diff
@@ -221,7 +221,14 @@ impl Man {
 
     fn _render_synopsis_section(&self, roff: &mut Roff) {
         roff.control("SH", ["SYNOPSIS"]);
-        render::synopsis(roff, &self.cmd);
+        if self.cmd.is_flatten_help_set() {
```

Comment:
Please add a test for this in https://github.com/clap-rs/clap/blob/master/clap_mangen/tests/testsuite/roff.rs

As mentioned in our contrib documentation, please split this into two commits
1. A commit that adds the test using `flatten_help` that passes, showing the current behavior
2. A commit that adds support for `flatten_help` and updates the test snapshot to show the new behavior

Reply:
Done